components/brave_news/rust/rust_strip_html_unittest.cc

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/brave_news/rust/lib.rs.h"
+
+#include "third_party/googletest/src/googletest/include/gtest/gtest.h"
+
+std::string Strip(const std::string& input) {
+  return static_cast<std::string>(brave_news::strip_html(input));
+}
+
+TEST(RustStripHtml, Noop) {
+  EXPECT_EQ("Foo", Strip("Foo"));
+}
+
+TEST(RustStripHtml, CanStripTags) {
+  EXPECT_EQ("Foo", Strip("<p>Foo</p>"));
```

Comment:
These tests all seem to be ASCII, so you don't have coverage for your use of `chars()`. Maybe add some of the test cases from voca_rs?

```C
"á<!",
">天地不仁<",
"\u{00a0}<!",
"<SCRIPT>Ω≈ç≈≈Ω</SCRIPT>",
"<SCRIPT a=\"blah\">片仮名平仮名</SCRIPT>",
"<span style=\"color: rgb(51, 51, 51); font-family: \" microsoft=\"\" yahei=\"\" stheiti=\"\" wenquanyi=\"\" micro=\"\" hei=\"\" simsun=\"\" sans-serif=\"\" font-size:=\"\" 16px=\"\">】มีมี่’ เด็กสาวที่นอนไม่ค่อยหลับเนื่องจากกลัวผี ขี้เหงา และอะไรหลายๆ อย่างทำให้เธอมึนได้โล่เพราะไม่ค่อยได้นอน การที่เธอ นอนไม่หลับทำให้เธอได้เจอกับ ‘ดีเจไททัน’ แห่งคลื่น 99.99 MHzเขาจัดรายการในช่วง Midnight Fantasy ตีสามถึงตีห้า และมีมี่ก็เป็นผู้ฟังเพียงคนเดียวของเขาจากที่ตอนแรกเธอฟังดีเจไททันเพื่อช่วยปลอบประโลมการที่เธอต้องมาอยู่หอเพียงลำพัง แต่ไปๆ มาๆกลับกลายเป็นว่าเธออยู่รอฟังเขาทุกคืนทำให้เธอไปเรียนแบบมึนๆ จนบังเอิญไปนอนหลับซบ ‘ธรรม’ผู้ชายจอมกวนที่บังเอิญมานอนให้เธอซบ! จนอาจารย์สั่งให้ไปทำรายงานคู่กัน และนั่นก็เป็นที่มาของการที่เธอเริ่มไม่แน่ใจแล้วว่าเธอปลื้มดีเจไททัน หรือแอบหวั่นไหวกับนายจอมกวนคนนี้กันแน่</span><br />"
```

Reply:
Added their complete suite for this, figured we might as well (it's MIT licensed).

Reply:
@fmarier is this enough to require a copy of the MIT original license?

Reply:
Whether or not it's technically required, I'd say it's a good idea to include the license so that we give proper attribution to the author of the tests.

Reply:
@fallaciousreasoning Maybe this could just be a comment header in the unittest file, saying "Some test data from the volca_rs project. [paste LICENSE.md from volca_rs]"?